runhouse/resources/folders/s3_folder.py

```diff
@@ -124,3 +117,130 @@ def _to_data_store(
+                "run the following: gcloud init; "
+                "gcloud auth application-default login"
+            )
+
+        session = boto3.Session()
+
+        aws_credentials = session.get_credentials().get_frozen_credentials()
+
+        storage_account = (
+            storagetransfer.googleServiceAccounts().get(projectId=project_id).execute()
+        )
+
+        S3Folder.add_bucket_iam_member(
+            gs_bucket_name,
+            "roles/storage.admin",
+            "serviceAccount:" + storage_account["accountEmail"],
+            project_id=project_id,
+        )
+
+        transfer_job = {
+            "description": f"Transferring data from S3 Bucket \
+            {s3_bucket_name} to GCS Bucket {gs_bucket_name}",
+            "status": "ENABLED",
+            "projectId": project_id,
+            "transferSpec": {
+                "awsS3DataSource": {
+                    "bucketName": s3_bucket_name,
+                    "awsAccessKey": {
+                        "accessKeyId": aws_credentials.access_key,
+                        "secretAccessKey": aws_credentials.secret_key,
+                    },
+                },
+                "gcsDataSink": {
+                    "bucketName": gs_bucket_name,
+                },
+            },
+        }
+
+        response = storagetransfer.transferJobs().create(body=transfer_job).execute()
+
+        operation = (
+            storagetransfer.transferJobs()
+            .run(jobName=response["name"], body={"projectId": project_id})
+            .execute()
+        )
+        logger.info(
+            f"Transfer job scheduled: s3://{s3_bucket_name} -> gs://{gs_bucket_name}"
+        )
+
+        logger.info("Waiting for the transfer to finish")
+
+        timeout = False
+        start = time.time()
+
+        while True:
+            # Get the status of the transfer operation
+            result = (
+                storagetransfer.transferOperations()
+                .get(name=operation["name"])
+                .execute()
+            )
+
+            if "error" in result:
+                raise RuntimeError(result["error"])
+
+            if "done" in result and result["done"]:
+                logger.info(
+                    f"Transfer finished in {(time.time() - start) / 60:.2f} minutes."
+                )
+                break
+
+            # Check if the elapsed time exceeds the timeout
+            if (time.time() - start) > TIMEOUT_SECONDS:
+                timeout = True
+                break
+
+            time.sleep(POLL_INTERVAL)
+
+        if timeout:
+            logger.info(
+                f"Transfer timed out after {(time.time() - start) / TIMEOUT_SECONDS:.2f} "
+                "hours. Please check the status of the transfer job in the GCP "
+                "Storage Transfer Service console at "
+                "https://cloud.google.com/storage-transfer-service"
+            )
+
+    @staticmethod
+    def add_bucket_iam_member(
```

Comment:
is this relevant to s3 or just google cloud?

Reply:
oh good point its just relevant for GCS, will move into the google folder's class